File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/service/impl/LoginServiceImpl.java
package com.alpha.ogo.service.impl;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import com.alpha.ogo.domain.User;
import com.alpha.ogo.repo.UserRepo;
import com.alpha.ogo.security.MyUserDetails;


@Service
public class LoginServiceImpl implements UserDetailsService{
	
	@Autowired
    UserRepo userRepository;
	
	@Override
    public MyUserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
        //return new User("foo", "foo",new ArrayList<>());
		Optional<User> user = userRepository.findByPhoneNo(userName);
        user.orElseThrow(() -> new UsernameNotFoundException("Not found: " + userName));
        return user.map(MyUserDetails::new).get();
        
    }

}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/service/impl/LoginServiceImpl.kt
package com.allion.ogoss.service.impl

import com.allion.ogoss.domain.User
import com.allion.ogoss.mapper.UserMapper
import com.allion.ogoss.security.MyUserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import java.util.function.Supplier


/**
 * @author: Tharindu
 * @Date: 6/8/2021
 * @Time: 4:03 PM
 */
@Service
class LoginServiceImpl(var userRepository : UserMapper) : UserDetailsService {


    override fun loadUserByUsername(userName:String) : MyUserDetails {
        val user = userRepository.findByPhoneNo(userName);
        user.orElseThrow(Supplier { UsernameNotFoundException("Not found: $userName") })
        return MyUserDetails(user.get());
    }

    fun getUserDetails(userName:String) : User {
        return userRepository.findByPhoneNo(userName).get();
    }
}